[Workers] streams revision (#2741)
cloudflare/cloudflare-docs: products/workers/src/content/learning/using-streams.md

The [Streams API](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API) is a web standard API that allows JavaScript to programmatically access and process streams of data.

Workers scripts do not need to prepare an entire response body before delivering it to `event.respondWith()`. You can use [`TransformStream`](/runtime-apis/streams/transformstream) to stream a response body after sending the front matter (that is, HTTP status line and headers). This allows you to minimize:

- The visitor’s time-to-first-byte.
- The buffering done in the Workers script.

Minimizing buffering is especially important for processing or transforming response bodies larger than the Workers script memory limit. For these cases, streaming is the only implementation strategy.

<Aside type="note">

> By default, the Cloudflare Workers service streams. Only use these APIs for modifying the response body while maintaining streaming behavior. If your Workers script only passes subrequest responses back to the client verbatim without reading their body text, then its body handling is already optimal and you do not have to use these APIs.

Reviewer: Cloudflare Workers is capable of streaming responses using the [Streams APIs](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API)

> By default, the Cloudflare Workers service streams. Only use these APIs for modifying the response body while maintaining streaming behavior. If your Workers script only passes subrequest responses back to the client verbatim without reading their body text, then its body handling is already optimal and you do not have to use these APIs.

Reviewer: ```suggestion
By default, Cloudflare Workers is capable of streaming responses using the [Streams APIs](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API). Only use these APIs for modifying the response body while maintaining streaming behavior. If your Worker only forwards subrequest responses to the client verbatim without reading their body text, then its body handling is already optimal and you do not have to use these APIs.
```

> By default, Cloudflare Workers is capable of streaming responses using the [Streams APIs](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API). Only use these APIs for modifying the response body while maintaining streaming behavior. If your Worker only forwards subrequest responses to the client verbatim without reading their body text, then its body handling is already optimal and you do not have to use these APIs.

Reviewer 2: ```suggestion
By default, Cloudflare Workers is capable of streaming responses using the [Streams APIs](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API). To maintain the streaming behavior, you should only modify the response body using the methods in the Streams APIs. If your Worker only forwards subrequest responses to the client verbatim without reading their body text, then its body handling is already optimal and you do not have to use these APIs.
```
Nit: MDN calls it the Streams API (singular), but we call it Streams APIs (plural).

<Aside type="warning" header="Warning">

The Streams API is only available inside of the [Request context](/runtime-apis/request), inside the `fetch` event listener callback.

</Aside>

--------------------------------

## Related resources

- [Streams API Reference](/runtime-apis/streams)

> - [MDN’s Streams API documentation.](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API)

Reviewer 2: ```suggestion
- [MDN’s Streams API documentation](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API)
```
